lib/liquid/template.rb

```diff
@@ -108,7 +108,7 @@ def initialize
     def parse(source, options = {})
       parse_context = configure_options(options)
 
-      unless source.valid_encoding?
+      if source.is_a?(String) && !source.valid_encoding?
```

Comment:
What if the `to_s` value is a string with an invalid encoding?

Reply:
Good catch... That is possible... 
```ruby
class ByteString
  def to_s
    "\xC0"
  end
end

Liquid::Template.parse(ByteString.new) # this should raise an error
```

I have updated the PR to convert the `source` to a `String` object from the `Template#parse` function instead of `Tokenize#initialize`.